R/backport_linter.R

```diff
@@ -7,13 +7,37 @@
 #' @seealso [linters] for a complete list of linters available in lintr.
 #' @export
 backport_linter <- function(r_version = getRversion()) {
-  Linter(function(source_file) {
-    if (inherits(r_version, "numeric_version")) r_version <- format(r_version)
-    if (r_version < "3.0.0") {
-      warning("It is not recommended to depend on an R version older than 3.0.0. Resetting 'r_version' to 3.0.0.")
-      r_version <- "3.0.0"
+  if (is.character(r_version) &&
+    re_matches(r_version, rex(start, "release" %or% list("oldrel", maybe("-", digits)) %or% "devel", end))) {
+    # Support devel, release, oldrel, oldrel-1, ...
```

Comment:
are these "flavors" documented somewhere? if so maybe link that in a comment

Reply:
Not really. I found them in `r-lib/actions` which uses an API from r-hub to get them.